src/main/java/oxgame/app/Setup.java

```diff
@@ -14,32 +14,36 @@ public Setup(Supplier<String> input, Consumer<String> output) {
     }
 
     public void initializeAGame() {
-        Player firstPlayer = null;
-        try {
-            firstPlayer = Player.playerCreator(input,output);
-        } catch (IllegalArgumentException e) {
-            output.accept(e.getMessage()+'\n');
-            initializeAGame();
-        }
+        Player firstPlayer = Player.playerCreator(input,output);
         Player secondPlayer = Player.playerCreator(input,output,firstPlayer);
 
         RoundBuffer playerBuffer = new RoundBuffer();
         playerBuffer.addPlayer(firstPlayer);
         playerBuffer.addPlayer(secondPlayer);
-
-        currentState = new InitialState(playerBuffer,output);
+        board = Board.newBoard(5,5);
```

Comment:
this.board is a private field that seems to be used only in this method. It does not need to be a filed

Reply:
Corrected.